apollo-client/src/main/java/com/ctrip/framework/apollo/spring/boot/ApolloApplicationContextInitializer.java

```diff
@@ -120,6 +120,10 @@ public void initialize(ConfigurableApplicationContext context) {
    * @param environment
    */
   protected void initialize(ConfigurableEnvironment environment) {
+      
+    // should always initialize system properties like app.id in the first place
```

Comment:
```suggestion
    // need to initialize system properties like app.id again in case they are configured in external data sources like spring cloud config
```